app/worker/utils.go

```diff
@@ -146,6 +147,36 @@ func processJobJSON(data interface{}, workflow *model.Workflow, job *model.Job)
 		return result
 	} else if value, ok := data.(string); ok {
 		return processJobString(value, workflow, job)
+	} else if value, ok := data.(primitive.D); ok {
+		result := make(map[string]interface{})
+		for key, item := range value.Map() {
+			result[key] = processJobJSON(item, workflow, job)
+		}
+		return result
+	} else if value, ok := data.([]primitive.D); ok {
+		result := make([]interface{}, len(value))
+		for i, item := range value {
+			result[i] = processJobJSON(item, workflow, job)
+		}
+		return result
+	} else if value, ok := data.(primitive.M); ok {
+		result := make(map[string]interface{})
+		for key, item := range value {
+			result[key] = processJobJSON(item, workflow, job)
+		}
+		return result
+	} else if value, ok := data.([]primitive.M); ok {
+		result := make([]interface{}, len(value))
+		for i, item := range value {
+			result[i] = processJobJSON(item, workflow, job)
+		}
+		return result
```

Comment:
I believe these are impossible cases:
> 5. When unmarshalling, a field of type interface{} will follow the D/M type mappings listed above. BSON documents unmarshalled into an interface{} field will be unmarshalled as a D.
https://pkg.go.dev/go.mongodb.org/mongo-driver/bson@v1.4.3#hdr-Structs

Reply:
Ok, leaving them out and focusing only on primitive.D